crates/polars-plan/src/logical_plan/hive.rs

```diff
@@ -30,77 +30,57 @@ fn separator(_url: &Path) -> char {
 }
 
 impl HivePartitions {
-    pub fn get_statistics(&self) -> &BatchStats {
-        &self.stats
+    /// Constructs a new [`HivePartitions`] from a schema reference.
+    pub fn from_schema_ref(schema: SchemaRef) -> Self {
+        let column_stats = schema.iter_fields().map(ColumnStats::from_field).collect();
+        let stats = BatchStats::new(schema, column_stats, None);
+        Self { stats }
     }
 
-    /// Parse a url and optionally return HivePartitions
-    pub(crate) fn parse_url(url: &Path) -> Option<Self> {
-        let sep = separator(url);
+    /// Constructs a new [`HivePartitions`] from a path.
+    ///
+    /// Fails if the path does not contain any Hive partitions.
+    /// TODO: Read multiple paths, infer schema from first X paths.
```

Comment:
Yes, but I don't think that will be done here.

Reply:
Agreed - removed the TODO here.